### DISCUSSION TITLE
Abaqus

### DISCUSSION THREAD

**Discussion by Nikhil23121998** [peridigm/peridigm#302] [Q&A] [answered]
Is it possible to model a sample in Abaqus for peridigm simulation?

**Comment by oldninja** [comment, marked as answer]
I think you may be able to do an equivalent model. I recall a study from Dr. Silling [https://www.osti.gov/servlets/purl/1725746](https://www.osti.gov/servlets/purl/1725746)...

**Reply by Nikhil23121998** [reply]
I want to model laminate with different stacking sequence or fiber orientation, and I found it difficult in coreform cubit. Do you have any suggestion?

**Comment by djlittl** [comment]
It would take some work, but in theory you could convert an Abaqus mesh file to the exodus file format and then read it into Peridigm.  Peridigm can read 4-node tet elements and 8-node hex elements.  Peridigm converts them internally to meshfree particles.  Peridigm cannot read shell elements or anything other than hex, tet, and sphere elements.  You would have to dig into the Abaqus file format and write a Python script (or similar) to convert to exodus.  The exodus.py script would be useful for this.  I think this would take 1-2 weeks to get working, so it may not be worth it unless you have many Abaqus meshes to convert.

**Reply by Nikhil23121998** [reply]
Is it possible to model laminate with different stacking sequence with the help of coreform cubit?

**Comment by johntfoster** [comment]
The Python [meshio](https://pypi.org/project/meshio/) library can 
convert between Abaqus and Exodus formats.
